rmgpy/molecule/kekulize.pyx

```diff
@@ -0,0 +1,290 @@
+# encoding: utf-8
+# cython: linetrace=True
+
+################################################################################
+#
+#   RMG - Reaction Mechanism Generator
+#
+#   Copyright (c) 2009-2011 by the RMG Team (rmg_dev@mit.edu)
+#
+#   Permission is hereby granted, free of charge, to any person obtaining a
+#   copy of this software and associated documentation files (the 'Software'),
+#   to deal in the Software without restriction, including without limitation
+#   the rights to use, copy, modify, merge, publish, distribute, sublicense,
+#   and/or sell copies of the Software, and to permit persons to whom the
+#   Software is furnished to do so, subject to the following conditions:
+#
+#   The above copyright notice and this permission notice shall be included in
+#   all copies or substantial portions of the Software.
+#
+#   THE SOFTWARE IS PROVIDED 'AS IS', WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+#   FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+#   DEALINGS IN THE SOFTWARE.
+#
+################################################################################
+
+"""
+This module contains functions for kekulization of a aromatic molecule.
+
+The basic algorithm is as follows:
+1. Identify all aromatic rings in the molecule, based on bond types.
+2. For each ring, identify endocyclic and exocyclic bonds.
```

Comment:
I am a little confused about the usage of endocyclic and exocyclic in this file. [this source](http://encyclopedia2.thefreedictionary.com/endocyclic+double+bond) mentions that endocyclic are always in a ring. This file seems to state that benzene bonds can be exocyclic, which seems contradictory to the source. Maybe I am mistaken about the code functionality, or maybe you're using a different definition (like part of 2 rings). Eitherway,  it might be good to elaborate on this.